pkg/sql/drop_view.go

```diff
@@ -138,39 +136,80 @@ func descInSlice(descID descpb.ID, td []toDelete) bool {
 	return false
 }
 
-func (p *planner) canRemoveDependentView(
+func (p *planner) canRemoveDependentGeneric(
```

Comment:
i don't see why this and canRemoveDependent both need to exist. just in-line this function into that one.

Reply:
oh, I don't like it as well, but there're use cases where the depended-on is not an table descriptor for example a `column`...

Reply:
okay, I see, can you use commentary to make the distinction clearer? Honestly, it feels like you should call the below one `canRemoveDependentFromTable` and just call this one `canRemoveDependent`?

Reply:
good idea. changed.